### DISCUSSION TITLE
How to merge pores and retain imformation

### DISCUSSION THREAD

**Discussion by news3083** [PMEAL/OpenPNM#2874] [Q&A]
Hello!

I am interested in extracting networks from porous materials and using this module. As mentioned in a previous QA, the SNOW algorithm confirms that the throat diameter is larger than the pore diameter. I want to extract only pores that are larger than the throat. However, when I attempted to merge pores when the throat diameter was larger than the pore's, I found that the diameter and volume information becomes 'nan'. I would like to preserve the original information (such as throat cross-sectional area and equivalent diameter) before merging. Is there any way other than saving the information before merging and reallocating values manually?

**Comment by jgostick** [comment]
There is no "one size fits all" approach to this so we can't do it automatically because whatever we chose to do will help some cases and break others, so the only option is to leave it up to the users.  When you merge two pores into a single new one, what you are doing is essentially deleting the first two pores from the network, *then* adding a new one.  In terms of our internal data format, which you can picture as a spreadsheet, deleting two pores is like removing two rows from the spreadsheet and adding a new pore is like appending a new row to the end.  So, when adding the new row the question is "what values should we put into the columns" for this new row.  The volume *maybe* could be the sum of the deleted ones, but what about the diameter? It is probably not the sum of the deleted pores.  So may point is that we cannot assume what users intend, so you must do it yourself.  In you case you'll want to store the information from the pores you're merging, do what computation you're after, then insert that result into the new pore.

**Comment by news3083** [comment]
Thank you for your prompt response. I appreciate your guidance, and as you suggested, I will endeavor to tackle this independently.
